src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private Object[] values;
+    private int size;
+    private T oldValue;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds " + index);
+        }
 
+        grow();
+
+        if (index == size) {
+            values[size++] = value;
+        } else {
+            System.arraycopy(values, index, values, index + 1, size - index);
+            values[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        grow();
 
+        for (int i = 0; i < list.size(); i++) {
+            T element = list.get(i);
+            add(element);
+        }
```

Comment:
Method Add() already has the grow(), so it's redundant. 
```suggestion
        for (int i = 0; i < list.size(); i++) {
            T element = list.get(i);
            add(element);
        }
```